engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -1962,7 +1962,25 @@ namespace dmGameSystem
                 ReHash(&component);
             }
 
-            const Vector4 trans = component.m_World.getCol(3);
+            Matrix4 world_transform = component.m_World;
+            Vector4 trans;
+            if (CanUseQuads(&component.m_TexturesData) && component.m_TexturesData.m_NumTextures > 0)
+            {
+                    float pivot_x = 0.f;
+                    float pivot_y = 0.f;
+                    const dmGameSystemDDF::SpriteGeometry* geometry = component.m_TexturesData.m_Geometries[0];
+                    if (geometry)
+                    {
+                        pivot_x = geometry->m_PivotX;
+                        pivot_y = geometry->m_PivotY;
+                    }
+
+                    trans = world_transform * Point3(-pivot_x, -pivot_y, 0.0f);
```

Comment:
Main fix. I looked that `m_WorldsPosition` in `RenderListEntry` used only for sorting and for visibility check so it's ok to set `m_WorldPosition` to value with pivot points.
Let me know if it's not ok.

Reply:
Structurally, it feels a bit off, since the other application of the pivot is for vertex calculations, but here a special case is added on the object's world position.

So now we apply pivot in "two" places.

Also, the pivot isn't about changing the position of the component, but shifting the visual representation.
So this change should go into the corresponding place for the vertex calculation.